### DISCUSSION TITLE
Add data to existing model / extend model

### DISCUSSION THREAD

**Discussion by tgmaxson** [mir-group/nequip#243] [General]
I am looking into ways to train models faster and more accurately and had the question of is it possible to add training data to an existing model / dataset and continue training?  Using the "append: True" option for anything that changes the model in a significant way seems to throw an error, but I think this should be possible?  In a similar vein, is it possible to expand the NN layers of an existing NN or is it better to just always retrain from the start again?  

I just realized this is a [partial duplicate](https://github.com/mir-group/nequip/discussions/235), however the idea of expanding the NN doesn't seem to be.

**Comment by Linux-cpp-lisp** [comment]
Hi @tgmaxson ,

> Using the "append: True" option for anything that changes the model in a significant way seems to throw an error, but I think this should be possible?

Yes, `append` is for continuing the _same_ training run; `initialize_from_state` (the link you gave) is for starting a new training run with some change (such as new data) on an existing model state.

>  In a similar vein, is it possible to expand the NN layers of an existing NN or is it better to just always retrain from the start again?

This is sort of possible, but I've never done it. The main issue would be that for an $N$ layer NequIP / Allegro network, there are many tensor product "paths" that cannot lead to a scalar and thus do not contribute to the prediction. If you were to build a $>N$ layer network, you want those paths from the $N$-layer network since they _could_ contribute to the final scalar prediction of the $>N$ layer network.  Of course practically this also makes it a pain to deal with.

I think a simpler approach, if I'm understanding what you want, is to train your full $>N$ layer network, and then remove/reinitialize the final few layers and do your training with your new data.
